src/components/selection-list/list-navigation.ts

```diff
@@ -0,0 +1,44 @@
+import keycode = require('keycode');
+import {clamp} from '../../utils';
+import {getFocusableItemValues, ItemValue} from './basic-list';
+
+export class ListNavigation {
+    private values: ItemValue[] = [];
+    private position: number = -1;
+
+    constructor(listItems: React.ReactNode[]) {
+        this.values = getFocusableItemValues(listItems);
+    }
+
+    public getFocusedValue() {
+        return this.values[this.position];
```

Comment:
In this component we're using undefined to indicate no value (all the `=== undefined` checks), so this works as intended.